### ISSUE TITLE
bug: camel case option name behaves weirdly when combined with dashed option name

### ISSUE THREAD

**Issue by antoniom** [yargs/yargs#920]
Consider the following code:
```js
let args = require('yargs')
        .option('appName', {
                alias: 'app-name',
                describe: 'App name',
                choices: ['app1', 'app2'],
                type: 'string',
                default: 'app1'
        })
        .config({appName: 'app2'})
        .argv;

console.log(args);
```
When running `node index.js` the output generated is this:
```
{ 
  _: [],
  appName: [ 'app2', 'app2' ],
  'app-name': 'app2',
  '$0': 'index.js' 
}
```
while the expected one is:
```
{ 
  _: [],
  appName:  'app2',
  'app-name': 'app2',
  '$0': 'index.js' 
}
```

**Comment by bcoe** [comment]
@antoniom yargs gives you the camel case for free, so perhaps try this:

```
let args = require('yargs')
        .option('app-name', {
                alias: 'a',
                describe: 'App name',
                choices: ['app1', 'app2'],
                type: 'string',
                default: 'app1'
        })
        .config({appName: 'app2'})
        .argv;

console.log(args);
```

definitely confusing though! would appreciate help with a patch if you feel like getting involved, we have a community slack here if you're so inclined:

http://devtoolscommunity.herokuapp.com/

**Comment by antoniom** [comment]
@bcoe after some research, I found out that the problem is on yargs-parser.
I have opened an issue there and I will provide a PR also. I am gonna leave this one open, until a newer version of yargs-parser is integrated with yargs.

**Comment by bcoe** [comment]
@antoniom 👍 appreciate the patch.

**Comment by compwright** [comment]
This can be closed; the fix was released in yargs-parser@8.1.0.

**Comment by mleguen** [comment]
I confirm this is fixed.
